desktop-exporter/app/components/waterfall-view/waterfall-view.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useRef } from "react";
+import { FixedSizeList } from "react-window";
+import {
+  Text,
+  Flex,
+  Heading,
+  Spacer,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { useSize } from "@chakra-ui/react-use-size";
+
+import { SpanData } from "../../types/api-types";
+
+const waterfallItemHeight = 50;
+
+type WaterfallRowProps = {
+  index: number;
+  style: React.CSSProperties;
+  data: WaterfallViewProps;
+};
+
+function WaterfallRow({ index, style, data }: WaterfallRowProps) {
+  let { spans, selectedSpanID, setSelectedSpanID } = data;
+  let span = spans[index];
+
+  // Add zero-width space after forward slashes, dashes, and dots
+  // to indicate line breaking opportunity
+  let nameLabel = span.name.replaceAll("/", "/\u200B");
+  nameLabel = span.name.replaceAll("-", "-\u200B");
+  nameLabel = span.name.replaceAll(".", ".\u200B");
+
+  // Set the background colour to make the list striped.
+  let backgroundColour =
+    index % 2 ? "" : useColorModeValue("gray.50", "gray.700");
+  let borderLeft = "none";
+
+  //Set the style for the selected item
+  if (!!selectedSpanID && selectedSpanID === span.spanID) {
+    backgroundColour = useColorModeValue("pink.50", "pink.900");
+    borderLeft = "5px solid";
+  }
+
+  return (
+    <Flex
+      style={style}
+      bgColor={backgroundColour}
+      borderLeft={borderLeft}
+      borderColor="pink.500"
+      onClick={() => setSelectedSpanID(span.spanID)}
+    >
+      <Flex
+        width={244}
+        alignItems="center"
+        paddingStart={2}
+      >
+        <Text
+          noOfLines={2}
```

Comment:
I think we should probably keep this to one line and rely on overflow handling to add an ellipsis, but I'll let you decide